### PR TITLE
Adds support for the fcrepo.rebuild.on.start property which

### PR DESCRIPTION
**Pull request by dbernstein**
will cause the indices to be rebuilt on restart.

**JIRA Ticket**:  https://jira.lyrasis.org/browse/FCREPO-3611

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
Adds a new property that allows the user to force Fedora to reindex all content on restart.
# How should this be tested?
`mvn clean install -Pone-click`
1. Start Fedora and add an item via the UI.
2. Delete the corresponding OCFL object from the disk.
3. Verify that the object is still visible as a child of the repository root
4. Attempting to retrieving the link should result in an error.
5. Restart Fedora and verify the deleted links are still present
6. Restart Fedora again with `-Dfcrepo.rebuild.on.start=true`
7. Verify that the link to the resource is gone. (be sure to do a hard refresh of your browser)

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers

### CODE REVIEW

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/RebuildIT.java [hunk 1]
```diff
@@ -129,6 +142,59 @@ public void testRebuildOcfl() {
...
+
+        final var binaryId = FedoraTypes.FEDORA_ID_PREFIX + "/binary";
+        final var objectVersionFile = this.ocflRepository.getObject(ObjectVersionId.head(binaryId)).getFile("binary");
+        final var path = objectVersionFile.getStorageRelativePath();
+        //delete ocfl object from disk:  path is "<object root dir>/v1/content/binary" therefore resolve the great
+        //grandparent directory
+        final var ocflObjectDirectory =
+                Path.of(TARGET_FCREPO_HOME_DATA_OCFL_ROOT, path)
+                        .toFile().getParentFile().getParentFile().getParentFile();
+        FileUtils.deleteDirectory(ocflObjectDirectory);
```
Comment (pwinckles) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/RebuildIT.java:168]: Why not just do `ocflRepo.purgeObject(binaryId)`?

Comment (dbernstein) [reply fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/RebuildIT.java:168]: That's a good point.  I'll do that.

Comment (dbernstein) [reply fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/RebuildIT.java]: @pwinckles: this should be ready to go.  Also updated the documentation to reflect these changes.

**Review by pwinckles — COMMENTED** [review]
Haven't run the code yet, but it looks good apart from one minor comment

**Review by pwinckles — APPROVED** [review]
👍

**Pull request merged**